simulation/revolve2/simulation/scene/geometry/textures/_texture.py

```diff
@@ -0,0 +1,118 @@
+from abc import ABC
+from dataclasses import dataclass
+
+from ..._color import Color
+
+
+@dataclass
+class Texture(ABC):
+    """An abstract texture for mujoco models."""
+
+    _name: str
+    _primary_color: Color
+    _secondary_color: Color
+    _repeat: tuple[int, int]
+    _size: tuple[int, int]
+    _map_type: str
+    """
+    map_type represents how the texture is mapped to an object.
+    
+    - "2d": Maps the texture to a 2d surface.
+    - "cube": Wraps the texture around a cube object.
+    - "skybox": Like "cube" but maps onto the inside of an object. 
+    """
+
+    _specular: float  # ranges [0, 1]
+    _shininess: float  # ranges [0, 1]
+    _reflectance: float  # ranges [0, 1]
+    _emission: float
+
+    @property
+    def name(self) -> str:
```

Comment:
If you make your dataclass(frozen=True), then you get very similar behaviour without the need for adding getters